flink-python/pyflink/table/table_environment.py

```diff
@@ -715,6 +718,111 @@ def create_temporary_view(self, view_path, table):
+
+        .. note::
+
+            Please make sure the package file matches the platform and python version of cluster.
+            These packages will be installed using pip.
+
+        :param requirements_file_path: The path of "requirements.txt" file.
+        :type requirements_file_path: str
+        :param requirements_cache_dir: The path of the local directory contains all packages
+                                        listed in the "requirements.txt" file.
+        :type requirements_cache_dir: str
+        """
+        self._dependency_manager.set_python_requirements(requirements_file_path,
+                                                         requirements_cache_dir)
+
+    def add_python_archive(self, archive_path, target_dir):
+        """
+        Upload a packaged file to cluster. The file will be extracted to the working directory of
+        python UDF worker.
+
+        If the "target_dir" is specified. The file will be extracted to the directory with the
+        same name of the value of "target_dir".
+        The extracted path would be: ${python_udf_working_directory}/${target_dir}/
+
+        If the "target_dir" is not specified. The file will be extracted to the directory with the
+        same name of this archive file.
+        The extracted path would be: ${python_udf_working_directory}/{archive_file_name}/
+
+        If python UDFs need to run in a specific python environment and the environment is not
+        exist on cluster, this method can be used to upload the packaged environment.
+        Note that the python interpreter path of the uploaded environment should be specified
+        via the method :func:`pyflink.table.TableConfig.set_python_executable`.
+
+        The files uploaded via this method is also accessible in UDFs. Using relative path to access
+        them.
+
+        Example:
+        ::
+
+            # command executed in shell
+            # assert the relative path of python interpreter is py_env/bin/python
+            $ zip -r py_env.zip py_env
+
+            # python code
+            >>> table_env.add_python_archive("py_env.zip", "myenv")
+            >>> table_env.get_config().set_python_executable("myenv/py_env/bin/python")
+
+            # the files in archive is also accessible in UDF
+            >>> def my_udf():
+            ...     with open("myenv/py_env/data/data.txt") as f:
+            ...         ...
+
+        .. note::
+
+            Please make sure the uploaded python environment matches the platform that the cluster
+            running on and that the version is 3.5 or higher.
+
+        .. note::
+
+            Currently only .zip file format is supported.
+
+        :param archive_path: The archive file path.
+        :type archive_path: str
+        :param target_dir: The target dir name that the archive file extracted to.
+        :type target_dir: str
+        """
+        self._dependency_manager.add_python_archive(archive_path, target_dir)
```

Comment:
To check whether a file is zip-format we must read its actual content. If user specifies a DFS url the check will introduce additional IO. Suffix checking is also hard because there are too many file formats are actually zip format. Maybe rewriting the doc string in detail to tell users what file format are actually supported is a better choice?